src/Controller/ListPageRssController.php

```diff
@@ -393,33 +404,29 @@ protected function getChannelDescription(NodeInterface $node, CacheableMetadata
       return '';
     }
 
-    $facet_storage = $this->entityTypeManager->getStorage('facets_facet');
-
+    $facets = $this->getKeyedFacetsFromSource($list_source);
     // Load one of the source facets because we need to use it to determine the
     // current active filters using the query_string plugin.
-    $facet = $facet_storage->load(reset($available_filters));
+    $facet = reset($facets);
     $query_string = $this->urlProcessorPluginManager->createInstance('query_string', ['facet' => $facet]);
     $active_filters = $query_string->getActiveFilters();
-    // Load all the facets for the active filters.
-    /** @var \Drupal\facets\FacetInterface[] $facets */
-    $facets = $facet_storage->loadMultiple(array_keys($active_filters));
     // Determine the filter values for each of the active facets.
     foreach ($facets as $facet) {
       $field_definition = $this->getFacetFieldDefinition($facet, $list_source);
 
-      // The field definition can be NULL as well.
-      $keyed_definitions[$facet->id()] = $field_definition;
       $keyed_facets[$facet->id()] = $facet;
-      $active_filters_values[$facet->id()] = $active_filters[$facet->id()];
+      $active_filters_values[$facet->id()] = $active_filters[$facet->id()] ?? [];
     }
 
+    $active_filters_values = array_filter($active_filters_values);
+
     // Run through each of the active filter values and prepare their displays.
     foreach ($active_filters_values as $facet_id => $filters) {
-      $field_type = $keyed_definitions[$facet_id] ? $keyed_definitions[$facet_id]->getType() : NULL;
-      $id = $field_type ? $this->multiselectPluginManager->getPluginIdByFieldType($field_type) : NULL;
+      $facet = $keyed_facets[$facet_id];
+      $id = $this->multiselectPluginManager->getPluginIdForFacet($facet, $list_source);
```

Comment:
I made the `getPluginIdForFacet` method to contain all 3 methods of finding a relevant plugin:

* By the Drupal field type if the field definition can be determined
* By the search API data type
* By a very specific facet ID (most specific).